src/styles/index.scss

```diff
@@ -0,0 +1,101 @@
+/**
+ * Naming scheme for variables is `--component-name-size-property-modifier`.
+ */
+
+@import './variables.scss';
+
+:root {
+  /* Colors */
+  --color-primary: #{$ui-03};
+  --color-secondary: #{$ui-04};
+  --border-color-light: #dddddd;
+  --background-color-primary: #{$ui-background-01};
+  --background-color-secondary: #{$ui-background-02};
+  --text-color-primary: #{$black};
+
+  --gds-color-white: #{$white};
+  --gds-color-black: #{$black};
+  --gds-color-ui-background-01: #{$ui-background-01};
+  --gds-color-ui-background-02: #{$ui-background-02};
+  --gds-color-ui-01: #{$ui-01};
+  --gds-color-ui-02: #{$ui-02};
+  --gds-color-ui-03: #{$ui-03};
+  --gds-color-ui-04: #{$ui-04};
+  --gds-color-ui-05: #{$ui-05};
+  --gds-color-ui-06: #{$ui-06};
+  --gds-color-ui-07: #{$ui-07};
```

Comment:
I feel like these should be the reference values for all other variables. If you change them it cascades to all components. Or do you think it wouldn't work well in practice?

Reply:
I'm not really sure if we should turn these variables into sass variable... What would be the benefit? My feeling is that since all the components have been built so that they rely on the css variables, then adding another set of sass variables might make it more difficult to understand.

Reply:
Should work, I'll change to that and test it out.

Reply:
The benefit of having them available as sass variables is that we can use them programmatically where we need to. Otherwise if we ever have a case where we can't use css variables, we'll have to duplicate the color definition.

For example we can't iterate over css variables in WP when generating classes for text and background colors, but we can iterate over sass variables.

I don't see the downside, since they're not "turned into" sass variables, but rather the master definition is in the sass variable and that is then used everywhere where that color is needed, which is mostly the css variable.

Reply:
Let me know how to proceed, i.e. if I should remove all sass variables.

Reply:
> The benefit of having them available as sass variables is that we can use them programmatically where we need to. Otherwise if we ever have a case where we can't use css variables, we'll have to duplicate the color definition.

In a way I'd like to remove the sass variables too but there are definitely cases when it's useful to have. Mostly I'd like to remove them so we could just copy the content of this file into a project and modify the variables in one place.

For WP we can mostly do things like this:

```scss
$color-names:
  'white',
  'ui-background-01',
  'ui-background-02',
  'ui-01',
  'ui-02',
  'black',
  'ui-03',
  'ui-04',
  'ui-05',
  'ui-06',
  'ui-07';

@each $name in $color-names {
  .has-#{$name}-background-color {
    background-color: var(--gds-color-#{$name});
  }
}
```

But yeah there's definitely cases where you need access to the raw value too. At least earlier projects I used a lot of sass color functions but maybe they're not needed with GDS anymore–not sure. Third party libraries like foundation and others (I cant recall now) had to have raw values too but I guess that'll get better and we'll have fewer third party libraries.

Reply:
@oxyc @henritukiainen  Correct my if my thinking is wrong, so if you want to use web components such as gds-button directly in html, then the only way to override --button-background-color is css variables? Otherwise you would need to build the gds with the parent project since the web components are prebuilt? Or is it like this?